retry.go

```diff
@@ -0,0 +1,96 @@
+package main
```

Comment:
Please add the boilerplate copyright header.

Reply:
Done.